File: C++/844. Backspace String Compare.cpp
class Solution {
public:
    bool backspaceCompare(string s, string t) {
        vector<char> a, b;
        
        for(int i = 0; i < s.size(); i++) {
            if(s[i] == '#') {
                if (!a.empty()) {
                    a.pop_back();
                }
            } else {
                a.push_back(s[i]);
            }
        }
        
        for(int i = 0; i < t.size(); i++) {
            if(t[i] == '#') {
                if (!b.empty()) {
                    b.pop_back();
                }
            } else {
                b.push_back(t[i]);
            }
        }
        
        return a == b;
    }
};

File: JAVA/844. Backspace String Compare.java
class Solution {
    public boolean backspaceCompare(String s, String t) {
        List<Character> a = new ArrayList<>();
        List<Character> b = new ArrayList<>();
        for(Character c:s.toCharArray()){
            if(c=='#'){
                if(a.size()!=0){
                    a.remove(a.size()-1);
                }
            }
            else{
                a.add(c);
            }
        }
        for(Character c:t.toCharArray()){
            if(c=='#'){
                if(b.size()!=0){
                    b.remove(b.size()-1);
                }
            }
            else{
                b.add(c);
            }
        }
        return a.equals(b);
    }
}